src/components/MissionCard/MissionCard.js

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Spinner } from 'react-bootstrap';
+import {
+  getMissions,
+  joinMission,
+  leaveMission,
+  selectMission,
+} from '../../redux/Mission/MissionSice';
+import './MissionCard.css';
+
+function MissionCard() {
+  const dispatch = useDispatch();
+  const missions = useSelector(selectMission);
+  console.log(missions);
+
+  useEffect(() => {
+    dispatch(getMissions());
+  }, [dispatch]);
+
+  const handleToggleMission = (missionId, reserved) => {
+    if (reserved) {
+      dispatch(leaveMission(missionId));
+    } else {
+      dispatch(joinMission(missionId));
+    }
+  };
+
+  return (
+    <div>
+      {missions.isLoading ? (
+        <Spinner />
+      ) : (
```

Comment:
- [ ] The same thing can also be  noticed in the Mission page  when it comes to this component  👍 . Whenever the Missions  page is refreshed ,  the state is reset , Kindly ensure that once the user refresh the Mission page , the state does not have to be reset 😄 

### Before Re-rendered 

![Capture](https://github.com/GhostEsso/SpaceTravelersHub/assets/80458199/b2933d18-22ae-478f-9378-a96891c30012)


### After Re-rendered 

![Capture1](https://github.com/GhostEsso/SpaceTravelersHub/assets/80458199/312da65f-3d54-4589-8438-33b5e1f86d7e)

Noticed that the `join mission` state has changed 👍 , Kindly fix that